text/0000-road-to-typescript.md

```diff
@@ -0,0 +1,160 @@
+    - coordination (e.g. editions aren't done till docs also goes for TS and types)
+    - [ ] blueprints?
+    - [ ] more?
+
+## Detailed design
+
+The primary goal of this RFC is to add TypeScript as an officially supported language. This involves four major elements:
+
+- Identifying the [constraints](#constraints) under which adoption can proceed
+- Making explicit the [non-goals](#non-goals) of this project
+- [Defining official support](#official-support): what it does and does not entail
+- Building a [roadmap](#roadmap) for completing the effort
+
+### Constraints
+
+The following are our hard constraints in supporting TypeScript with Ember:
+
+- We must not compromise our commitment to “stability without stagnation” and our strong Semantic Versioning guarantees. 
+- Using TypeScript should never be mandatory for anyone who uses Ember.
+- TypeScript support must never *degrade* the experience of JavaScript users, and wherever possible it should benefit both JavaScript and TypeScript developers.
+
+### Non-Goals
+
+The combination of constraints and ecosystem status lead us to the following non-goals for this effort:
+
+#### Superseding or replacing JavaScript
+
+This RFC aims to *add* a new first-class language to the Ember support matrix. However, it does *not* recommend replacing JavaScript with TypeScript. To the contrary: this is an explicit non-goal. Per the constraints described above, adding first-class TypeScript support should be a net positive for JavaScript-only Ember developers.
+
+This is not only a non-goal of this RFC, but in fact should be taken as a normative non-goal for Ember itself; we do not foresee this changing in the future.
+
+#### Typed Templates
+
+This roadmap RFC explicitly does not require support for “typed templates” as part of the path to adopting TypeScript as a community. TypeScript-powered integration between the template layer and the JavaScript context is a key goal for the Typed Ember team already. However, it is not a *gating feature* for Ember to officially support TypeScript. Rather: the feature can be rolled out and integrated with Ember CLI, the Ember Language Server, and other tooling whenever it is ready, decoupled from the other efforts here. If we reach a point where full support for type-aware template integration exists and is sufficiently robust, we can consider adding it to our definition of official Ember support via a dedicated RFC.
+
+The *only* relationship those efforts have to this roadmap is that they will inform the design of the `@glimmer/component@2.x` TypeScript API (as described below under **Roadmap**).
+
+### Official Support
+
+<!--
+
+**What “official support” entails.** Does it require that all officially published libraries ship TypeScript types? Does it require a specific support policy for all officially published libraries? Should the RFC template be updated in any way to account for TypeScript API constraints? etc.
+
+-->
```

Comment:
- Libraries which are direct dependencies of the default blueprint and have public JavaScript APIs should ship their own types.
- Things like ember-template-lint (not an official library) may *not* need it:
     - not official
     - not a public JS API

Reply:
What are "officially published libraries" ?

Another idea from storybook team here: They grouped some of their addons as "essentials". Maybe adopting some strategy might help, too?

Reply:
I’ve clarified this in the latest revision. Can discuss further in the strike team meeting tomorrow!